examples/index.md

```diff
@@ -162,3 +162,241 @@ Content-Type: application/vnd.api+json
+      "detail": "First name must contain at least three characters."
+    },
+    {
+      "code":   "225",
+      "source": { "pointer": "data/attributes/password" },
+      "title": "Passwords must contain a letter, number, and punctuation character.",
+      "detail": "The password provided is missing a punctuation character."
+    },
+    {
+      "code":   "226",
+      "source": { "pointer": "data/attributes/password" },
+      "title": "Password and password confirmation do not match."
+    }
+  ]
+}
+```
+
+Notice that this response includes not only the `errors` top-level member,
+but the `jsonapi` top-level member. Error responses may not contain the
+top-level `data` member, but can include all the other top-level members
+JSON API defines.
+
+Also, notice that the third error object lacks a `detail` member (perhaps
+for security). Again, all error object members are optional.
+
+### Advanced `source` Usage
+
+In the example below, the user is sending an invalid JSON API
+request, because it's missing the `data` member:
+
+```http
+PATCH /posts/1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+
+{ "datum": [ ] }
+```
+
+Therefore, the server responds:
+
+```http
+HTTP/1.1 422 Unprocesssable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "source": { "pointer": "/" },
+      "detail":  "Missing `data` Member at document's top level."
+    }
+  ]
+}
+```
+
+It uses `source` to point to the top-level of the document (`"/"`).
+(Pointing to `"/data` would be invalid because the request document did
+not have a value at `"/data"`, and `source` is always given with reference
+to the request document.)
+
+Note that if the server cannot parse the request as valid JSON, including
+`source` doesn't make sense (because there's no JSON document for `source` to
+refer to). Here's how the server might respond to an invalid JSON document:
+
+```
+{
+  "errors": [{
+    "status": "400",
+    "detail": "JSON parse error - Expecting property name at line 1 column 2 (char 1)."
+  }]
+}
+```
+
+#### Invalid Query Parameters
+
+The `source` member can also be used to indicate that the error originated
+from a problem with a URI query parameter, like so:
+
+```http
+GET /api/posts/1?include=auther
+```
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "source": { "parameter": "include" },
+      "title":  "Invalid Query Parameter",
+      "detail": "The resource does not have an `auther` relationship path."
+    }
+  ]
+}
+```
+
+In most cases, JSON API requires the server to return an error when it encounters
+an invalid value for a JSON API–defined query parameter. However, for API-specific
+query parameters (i.e. those not defined by JSON API), a server may choose to
+ignore an invalid parameter and have the request succeed, rather than respond with
+an error. API-specific query paramters must contain one non a-z character.
```

Comment:
typo `parameter`, also, on first read, it's unclear to me what it means for API-specific query parameters to contain a non a-z character and I haven't yet found it in the `schema` or the format page.

? http://jsonapi.org/format/#document-member-names

> To enable an easy mapping of member names to URLs, it is RECOMMENDED that member names use only non-reserved, URL safe characters specified in RFC 3986.